cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java

```diff
@@ -121,6 +122,30 @@ public KafkaCruiseControlConfig(Map<?, ?> originals, boolean doLog) {
     return objects;
   }
 
+  /**
+   * Sanity check to ensure that {@link ExecutorConfig#ZOOKEEPER_CONNECT_CONFIG} is set if
+   * <ul>
+   *   <li>{@link AnomalyDetectorConfig#KAFKA_BROKER_FAILURE_DETECTION_ENABLE_CONFIG} is not enabled.</li>
+   *   <li>{@link MonitorConfig#TOPIC_CONFIG_PROVIDER_CLASS_CONFIG} is set to com.linkedin.kafka.cruisecontrol.config.KafkaTopicConfigProvider.</li>
+   * </ul>
+   */
+  private void sanityCheckZK() {
+    String zkConnect = getString(ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG);
+    if (zkConnect == null || zkConnect.isEmpty()) {
+      if (!getBoolean(AnomalyDetectorConfig.KAFKA_BROKER_FAILURE_DETECTION_ENABLE_CONFIG)) {
+        throw new ConfigException(String.format("Missing required configuration, either %s must be set, or %s must be set to true.",
+                                                ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG,
+                                                AnomalyDetectorConfig.KAFKA_BROKER_FAILURE_DETECTION_ENABLE_CONFIG));
+      }
+      if (KafkaTopicConfigProvider.class.getName().equals(getClass(MonitorConfig.TOPIC_CONFIG_PROVIDER_CLASS_CONFIG).getName())) {
+        throw new ConfigException(String.format("Missing required configuration, either %s must be set, or %s must be set to %s.",
+                ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG,
+                MonitorConfig.TOPIC_CONFIG_PROVIDER_CLASS_CONFIG,
+                KafkaAdminTopicConfigProvider.class.getName()));
```

Comment:
Nit:Bad indentation.
If the IDE supports it (e.g. IntelliJ), I'd recommend importing [the style file of CC](https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_5/docs/code-style.xml) to automatically handle such style issues.

Reply:
Thanks that's useful!